### PR TITLE
Fix for Tough Break Universal Switch speed buff breaking robots

### LINKED ISSUE #258: [MVM] Tough Break Universal Switch speed buff breaks robots
Ever since Gun Mettle's universal weapon switch buff (0.67 to 0.50), Bots with items such as Bonk! Atomic Punch or Soldier Banners will often attempt to switch to it, but will be stuck in a endless loop of switch as the AI calculates the old speed, causing the robot to never attack. Most noticeable with giant Banners on Rottenburg or giant bonks on Mannhattan.

### Checklist
- [x] None of the open *or* closed issues document this problem.
- [x] This is a bug, not a new idea.
- [x] This is objectively a bug, and not something that people commonly disagree on.
- [x] This bug is in the code of the game, not in the textures, sounds, or other resources.
- [x] This issue occurs on the latest vanilla TF2 update.
- [ ] This issue is listed on the [Source 1 official bugtracker](https://github.com/ValveSoftware/Source-1-Games/issues).
- [ ] This issue occurs with DirectX level 100.
- [x] This issue occurs on the latest Team Comtress release, or the `master` branch.
- [x] This issue occurs when using no custom content.
- [x] This issue occurs when using `start_tf2.bat` (or `start_server.bat`) with no custom launch options.
- [x] This issue occurs when using a listen server (using the `map` command ingame).
- [x] This issue occurs when using a dedicated server.
- [ ] This bug is in the internal procedures or tooling (e.g. the build process).

### Reproduction
1. Go to any valve wave with Banner/Bonk Giants (Bavarian Wave 2, or Empire Escalation pre final wave)
2. Spectate banner/bonk scout
3. Will randomly stop using it's weapon and begin infinitely switching back and forth

### PR DESCRIPTION
**Pull request by Thestickman391**
Closes #258

### CODE REVIEW

**Comment by Thestickman391** [comment]
Also closes #224

File: game/server/tf/bot/tf_bot.cpp [hunk 1]
```diff
@@ -4234,7 +4234,11 @@ Action< CTFBot > *CTFBot::OpportunisticallyUseWeaponAbilities( void )
 			CTFBuffItem *buff = (CTFBuffItem *)weapon;
 			if ( buff->IsFull() )
 			{
-				return new CTFBotUseItem( buff );
+				if (CTFBotUseBuffItem::IsPossible( reinterpret_cast<CTFBot *>( this ) ) ) {
```
Comment (maximsmol) [inline game/server/tf/bot/tf_bot.cpp:4237]: Why is the `reinterpret_cast` needed? Is `this` not of type `CTFBot*` already? If not, could a `dynamic_cast` work instead?

File: game/server/tf/bot/tf_bot.cpp [hunk 2]
```diff
@@ -4246,7 +4250,11 @@ Action< CTFBot > *CTFBot::OpportunisticallyUseWeaponAbilities( void )
 				// scout lunchboxes are also gated by their energy drink meter
 				if ( !IsPlayerClass( TF_CLASS_SCOUT ) || m_Shared.GetScoutEnergyDrinkMeter() >= 100 )
 				{
-					return new CTFBotUseItem( lunchbox );
+					if (CTFBotUseLunchBoxItem::IsPossible( reinterpret_cast<CTFBot *>( this ) ) ) {
```
Comment (maximsmol) [inline game/server/tf/bot/tf_bot.cpp:4253]: ditto

**Review by maximsmol — CHANGES_REQUESTED** [review]
Thanks for the patch! Take a second look at the `reinterpret_cast`s please

**Comment by Thestickman391** [comment]
You are correct - changes made

**Pull request closed without merging**